### DISCUSSION TITLE
How to change entry point after file was loaded?

### DISCUSSION THREAD

**Discussion by denys-volodarskyi** [Vector35/binaryninja-api#6247] [General] [answered]
I want to change entry and reanalyze from this address.

``bv.entry_point`` is read-only

**Comment by psifertex** [comment]
You cannot but also, there's no real reason to. The entry point is only special in that it is assumed to have a function there. Simply make another function anywhere else and the same recursive descent analysis will happen from that function as well. You can do this with "p" in the UI or [create_user_function](https://api.binary.ninja/binaryninja.binaryview-module.html#binaryninja.binaryview.BinaryView.create_user_function).

**Comment by denys-volodarskyi** [comment]
Thanks for the response.

My entry function defines R13 register (PowerPC) and it's global pointer used later in many other functions.

If I set it via Analysis/Global Pointer Value (ConstantPointerValue) and do Reanalyze it's propagated to these functions. But after restart it's gone and I have to set it manually again.

I thought the reason is analysis root function is not properly set and it cannot do it automatically.

**Reply by psifertex** [reply]
Ahh, that's a good question, let me check with my co-workers, I believe setting it via analysis should be maintained, that might be a separate bug.

**Reply by xusheng6** [reply]
1. It should be saved and this sounds like a bug. Can you share the file with us (in public or in private) so that I can reproduce it?
2. This has nothing to do with whether the function being the entry point function

**Comment by denys-volodarskyi** [comment]
That's powerpc binary

r13 is set at 0x202c8 as result of 

```
000202c8  3da04001   lis     r13, 0x4001
000202cc  39ad91e4   addi    r13, r13, -0x6e1c  {0x400091e4}
```

r2 seems to be global value too, but currently I guess there's no way to set more than one value

**Reply by xusheng6** [reply, marked as answer]
I have created the issue https://github.com/Vector35/binaryninja-api/issues/6253 to track this bug
